programs/token/src/state/mint.rs

```diff
@@ -0,0 +1,66 @@
+use pinocchio::{account_info::AccountInfo, pubkey::Pubkey};
+pub struct Mint(*const u8);
+
+impl Mint {
+    pub const LEN: usize = 82;
+
+    pub fn from_account_info_unchecked(account_info: &AccountInfo) -> Self {
+        unsafe { Self(account_info.borrow_data_unchecked().as_ptr()) }
+    }
+
+    pub fn from_account_info(account_info: &AccountInfo) -> Self {
```

Comment:
I am not sure about this pattern. The `borrow_data_unchecked` does not keep track of the borrows but this is not obvious from the method name `from_account_info`.

I wonder if this should use `try_borrow_data` instead, although I see that the `unchecked` in this case refers to the length and owner checks.

Reply:
I'm really not sure what good counting borrows does us when the only way to modify this segment of memory is by CPI, and CPIs already don't care about us pretending using memory we don't own is safe.

Reply:
Yeah, I see your point. I am thinking on the case of the `CloseAccount` instruction for example. You could get the pointer to the token account and then do a CPI to close the account. After the CPI, that pointer is technically invalid, since the account has been resized to `0`.

The "checked" version of the `invoke_signed`, which is the one that the instruction builder is using, checks the borrow flags: https://github.com/febo/pinocchio/blob/main/sdk/pinocchio/src/program.rs#L91-L97

So if you have an "active" borrow, you will get a `ProgramError::AccountBorrowFailed` error before the CPI. I think we should offer a method that allows to bypass the borrow check and let the client code decide when that is safe, but I wonder if this behaviour should be behind a method name that makes that explicit.